### DISCUSSION TITLE
GPS polling frequency for battery saving

### DISCUSSION THREAD

**Discussion by ZebNemeth** [peterLaurence/TrekMe#78] [Ideas]
For mountaineering, it is useful to set location update speed to once every twenty minutes or so, and upon the press (or tap) of a button. Perhaps include a slider in the settings menu for this? 
Greatly appreciate the software thus far! If this setting is included, you'd save many people the headache of acuiring a Garmin Fenix or other dedicated device. They'd not need to.

**Comment by p-lr** [comment]
Interesting idea, thanks. I'll think about it.

**Comment by p-lr** [comment]
For what is worth, TrekMe stops polling GPS locations when the screen is locked of when the app looses focus. It resumes polling only when the map is visible on screen.
So, IIUC, you'd like to limit the GPS polling rate even when the map is visible?

**Reply by ZebNemeth** [reply]
Aha! I have to admit I had only shortly tested and not noticed this behaviour. If it is as you say, that should fix my request for now.

I meant my idea to refer to polling as a background process.


So almost no battery is spent while the screen is locked, then?


Thanks for your timely reply, greetings from the Netherlands

Edit: this appears to not be the case. Polling continues after screen lock.

**Reply by ZebNemeth** [reply]
I have to say, your description of the polling behaviour does not match my experience. I've just tested it and it appears the polling keeps going when screen is locked.
The blue track is while looking at the screen, orange is while trying to get the GPS to stop polling upon locked screen, i.e. walking from one end of the garden to the other whilst the screen is locked. It matches my path very precisely, and the jittery orange bit is where I left my phone for a while to maybe get some timer to run out. Even after some five minutes of idleness, I picked up my phone and walked some more, and unlocked the screen some distance away, the path was recorded at what appears as a polling rate of once per second.![Screenshot_20210616-192834__01.jpg](https://user-images.githubusercontent.com/58953998/122266442-1676aa00-ceda-11eb-8d88-67632dd35980.jpg)

**Comment by p-lr** [comment]
Oh, sorry, I should have mentioned that this is not the case during a GPX recording.
During a recording, the GPS is constantly polled, and this is by design.
Otherwise, during a normal hike, and without GPX recording, the polling stops when the screen is locked.

**Reply by p-lr** [reply]
GPX recording is a "resource hungry" task, and it's not advised to do that when saving the battery is critical. Reducing the poll rate during the recording might be an interesting option though..
